app/controllers/administrateurs/procedures_controller.rb

```diff
@@ -194,21 +194,56 @@ def clone
     def archive
       procedure = current_administrateur.procedures.find(params[:procedure_id])
 
-      if params[:new_procedure].present?
-        new_procedure = current_administrateur.procedures.find(params[:new_procedure])
-        procedure.update!(replaced_by_procedure_id: new_procedure.id)
+      if procedure.update(closing_params)
+        procedure.close!
+        if (procedure.dossiers.not_archived.state_brouillon.present? || procedure.dossiers.not_archived.state_en_construction_ou_instruction.present?) && procedure.closing_reason == Procedure.closing_reasons.fetch('other')
+          redirect_to admin_procedure_closing_notification_path
+        else
+          flash.notice = "Démarche close"
+          redirect_to admin_procedure_path(id: procedure.id)
+        end
+      else
+        flash.alert = procedure.errors.full_messages
+        redirect_to admin_procedure_close_path
       end
 
-      procedure.close!
-
-      flash.notice = "Démarche close"
-      redirect_to admin_procedures_path
-
     rescue ActiveRecord::RecordNotFound
       flash.alert = 'Démarche inexistante'
       redirect_to admin_procedures_path
     end
 
+    def closing_notification
+      @procedure = current_administrateur.procedures.find(params[:procedure_id])
+      @users_brouillon_count = @procedure.dossiers.not_archived.state_brouillon.map(&:user).uniq.count
+      @users_en_cours_count = @procedure.dossiers.not_archived.state_en_construction_ou_instruction.map(&:user).uniq.count
+    end
+
+    def notify_after_closing
+      @procedure = current_administrateur.procedures.find(params[:procedure_id])
+      @procedure.update!(notification_closing_params)
+
+      if (@procedure.closing_notification_brouillon? && params[:email_content_brouillon].blank?) || (@procedure.closing_notification_en_cours? && params[:email_content_en_cours].blank?)
+        flash.alert = "Veuillez renseigner le contenu de l’email afin d’informer les usagers"
+        redirect_to admin_procedure_closing_notification_path and return
+      end
+
+      if @procedure.closing_notification_brouillon?
+        users = @procedure.dossiers.not_archived.state_brouillon.map(&:user).uniq
```

Comment:
Optionnel, on peut modifier le code pour ne passer que les users_id

```suggestion
        user_ids = @procedure.dossiers.not_archived.state_brouillon.pluck(:user_id)
```

Reply:
:+1: